kitsune/questions/static/questions/js/components/AttachmentField.jsx

```diff
@@ -0,0 +1,64 @@
+/* globals React:false */
+import AAQStep from './AAQStep.jsx';
+import AAQActions from '../actions/AAQActions.es6.js';
+import {authStates} from '../../../users/js/constants/UserAuthConstants.es6.js';
+
+export default class AttachmentField extends AAQStep {
+  handleChange(ev) {
+    let data = new FormData();
+
+    let files = ev.target.files;
+
+    for (var file of files) {
+      AAQActions.uploadImage(file);
+    }
+  }
+
+  heading() {
+    return 'Add images';
+  }
+
+  body() {
+    if (this.props.userAuth.state === authStates.LOGGED_IN) {
+      return (
+        <div className="AAQApp__AttachmentField">
+          <input type="file" name="image" size="30" title="Browse for an image to upload."
+                 onChange={this.handleChange.bind(this)}/>
+          <ul className="AAQApp__AttachmentField__Attachments">
+            {this.props.question.images.map((image) => {
+              return (
+                <Attachment image={image}/>
+              );
+            })}
```

Comment:
By the way, this can be shortened to simply

```jsx
{this.props.question.images.map(image => <Attachment image={image}/>)}
```

ES6 arrow functions that simply return a single value immediately can drop the `{ return ...; }` parts of the construct.